colexec: begin refactor of colexec type system

yuzefovich opened the pull request:
**pkg/col: rename coltypes package to phystypes**

This commit renames col/coltypes to col/phystypes. We have "logical"
type system (sql/types) and "physical" type system (previously
col/coltypes). Later we will introduce "vectorized execution" type
system (maybe something like colexec/colexectypes) which will combine
two others. And the goal of this change is to reduce the amount of
confusion.

Release note: None

**colexec/typeconv: rename typeconv to colexectypes**

This commit renames colexec/typeconv to colexec/colexectypes which will
soon become the "vectorized execution type system" package.

Release note: None

**colexectypes: introduce new colexectypes.T struct**

This commit introduces a new "columnar execution" type that is a wrapper
around phystypes.T that also stores the original "logical" SQL type.

Addresses: #43559.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/43702)
<!-- Reviewable:end -->

yuzefovich committed:
pkg/col: rename coltypes package to phystypes

This commit renames col/coltypes to col/phystypes. We have "logical"
type system (sql/types) and "physical" type system (previously
col/coltypes). Later we will introduce "vectorized execution" type
system (maybe something like colexec/colexectypes) which will combine
two others. And the goal of this change is to reduce the amount of
confusion.

Release note: None

yuzefovich committed:
colexec/typeconv: rename typeconv to colexectypes

This commit renames colexec/typeconv to colexec/colexectypes which will
soon become the "vectorized execution type system" package.

Release note: None

yuzefovich committed:
colexectypes: introduce new colexectypes.T struct

This commit introduces a new "columnar execution" type that is a wrapper
around phystypes.T that also stores the original "logical" SQL type.

Release note: None

yuzefovich commented:
Closing this in favor of #43723.
